lib/core/client.js

```diff
@@ -145,6 +151,8 @@ class Client extends EventEmitter {
     this[kClosed] = false
     this[kDestroyed] = false
     this[kServerName] = null
+    this[kTLSMaxCachedSessions] = (tls == null || tls.maxCachedSessions == null) ? 100 : tls.maxCachedSessions
+    if (tls && tls.maxCachedSessions) delete tls.maxCachedSessions
```

Comment:
Is this the same as node core?

Reply:
Nope, in node core it's an agent option.

Reply:
We should not mutate options.

Reply:
Do you prefer a `tls` and `tlsMaxCachedSessions` options?

Reply:
Not sure why you do the delete?

Reply:
I can remove it, it was for not passing to the Node core tls implementation an option that should not be there :)
I can destructure the undici tls option if you would like to avoid mutating objects.